rosbag2_storage/src/rosbag2_storage/impl/storage_factory_impl.hpp

```diff
@@ -116,8 +116,9 @@ get_interface_instance(
     registered_classes.begin(),
     registered_classes.end(), storage_options.storage_id);
   if (class_exists == registered_classes.end()) {
-    ROSBAG2_STORAGE_LOG_WARN_STREAM(
-      "No storage plugin found with id '" << storage_options.storage_id << "'.");
+    // This should not print a warning, because it can be used by open_read_only twice,
+    // legitimately expecting to fail for READ_ONLY but succeed for READ_WRITE
+    // The extra output is misleading to end users.
```

Comment:
@emersonknapp May be still keep logging with redirection to the debug stream? In regular usage users will not see such message.
I am worry about a case when we can fail loading plugin in second iteration with `read_write` interface and will not have any meaningful outputs or logging about it.

Reply:
There is already an ERROR message on the second (or first) iteration in the public APIs. This function is an implementation internal 

https://github.com/ros2/rosbag2/blob/e948e954700335b326daee3152f81221e4b7d42a/rosbag2_storage/src/rosbag2_storage/impl/storage_factory_impl.hpp#L199

Reply:
Ok good. :+1: